conf/cassandra.yaml

```diff
@@ -1687,4 +1687,8 @@ drop_compact_storage_enabled: false
 #    enabled: true # (overriden by cassandra.ignore_dc system property)
 #  rack:
 #    enabled: true # (overriden by cassandra.ignore_rack system property)
-
+#  gc_grace_period:
```

Comment:
agreed. i don’t like this format of configuration since it’s not immediately clear to uses what each startup check does.

I think we should use the following format for startup checks to make it easier for users to reason about:

```
check_dc:
    enabled: true
check_rack:
    enabled: true
check_data_resurrection:
    enabled: true
    foo: bar
```

Reply:
I can document it more, sure. I dont feel like rewriting the names would help. This feels weird:

    startup_checks:
        check_dc:

We already established the fact that this section is about startup checks, what you need to have another "check_" in front of it for?

Reply:
Also, I think we can specify system properties for Cassandra like `-Dcassandra.startup_checks.dc.enabled=true`. That works way better than `-Dcassandra.startup_checks.check_dc.enabled=true`.

Reply:
“dc” does not convey to the reader of the configuration file what this does, while “check_dc” makes it immediately clear that the startup check will verify if the DC is valid.

I think it’s a good practice of readability and usability to prefix all startup checks with “check_something” instead of just “something” which does not provide enough context for the reader.

Reply:
users do not need to read documentation to understand what a startup check does, it should be straightforward to understand this from the check name:

check_dc: verify if dc is valid
check_rack: verify if rack is valid
check_data_resurrection: verify if there was data resurrection

Reply:
I dont think so, Paulo. We are repeating the information. Indeed it is about checking the dc, what it would be about if it is in startup_checks section? The context is the section it is in, that is why it is embedded in it.

Reply:
We already use the “check” prefix in the StartupChecks class. I don’t see a reason *not* to repeat this good practice here - to make startup checks easier to understand by people who are not familiar with it.

Reply:
“dc” is a noun while “check_dc” is a verb. Every startup check should be an action and not a contextless noun like “dc”. `-Dcassandra.startup_checks.check_dc.enabled=true` looks perfectly fine to me.

Reply:
That is false argument for this case because that is internal implementation and this is facing the user.

Reply:
what do you find more meaningful for property name @driftx : `check_dc` or `dc`? `gc_grace_seconds` or `check_data_resurrection`?

Reply:
> That is false argument for this case because that is internal implementation and this is facing the user.

Oh, so we should provide meaningful variable names for internal developers and cryptic configuration names for users?

Reply:
> users do not need to read documentation to understand what a startup check does, it should be straightforward to understand this from the check name:
> 
> check_dc: verify if dc is valid check_rack: verify if rack is valid check_data_resurrection: verify if there was data resurrection

Again you need to look at this from perspective - it is in some section called "startup_checks". That key "dc" or "rack" does not live alone, it is inherently connected to the section it is in.

Reply:
I slghtly prefer check_dc over dc, and strongly prefer check_data_resurrection vs gc_grace_seconds.

Reply:
> Again you need to look at this from perspective - it is in some section called "startup_checks". That key "dc" or "rack" does not live alone, it is inherently connected to the section it is in.

`startup_checks.dc` does not tell me what this startup check does while `startup_checks.check_dc` tells me without reading any documentation that this startup check will validate the datacenter

Reply:
> > That is false argument for this case because that is internal implementation and this is facing the user.
> 
> Oh, so we should provide meaningful variable names for internal developers and cryptic configuration names for users?

It is only your opinion it is cryptic, not mine. The internal names are for variables, it is used in a completely different context. How can you compare that?

Reply:
@driftx is a seasoned developer and was not able to understand what the property does - I’m sure users will have a hard time to understand what `startup_checks.gc_grace_seconds` does without reading additional documentation while `startup_checks.check_data_resurrection` quickly conveys the information without additional context

Reply:
For this specific check I am for sure open to changes, I was struggling to name it appropriately myself to catch what it is doing. However prefixing it with "check" is really questionable on my side tbh.

Reply:
> It is only your opinion it is cryptic, not mine. 

of course it’s not cryptic to you, you implemented the check. I’m thinking of external users and not myself.

Reply:
_startup_checks.dc does not tell me what this startup check does_

What is unclear here? What about it is checking a dc?

Reply:
> However prefixing it with "check" is really questionable on my side tbh.

Not prefixing startup check names with “check_” would go against the current conventions used in the “StartupChecks” class and would confuse users for sure and affect  usability so I’m against it.

Reply:
I think we are moving in circles here, I already explained to you that how variables are called does not have to do anything with how it stuff called in cassandra.yaml _at all_. I will not return to that anymore.

Reply:
I’m against merging this patch unless we prefix startup checks with “check_”. 😊

Reply:
Cassandra is already a very complicated database to configure. I don’t want to make life harder for users.

Reply:
@pauloricardomg , I am very sad to say it but I frankly think that your declared interest in the userbase is, least to say, phony. If your interest was genuine, you would not sabotage this ticket by your rather remarkably strong preference to include some prefix to a configuration key, even making fun of that. What is funny to you? 

That we are not more than two weeks from the freeze and you are sending me trolling faces in the private communication telling me your notebook has broken yesterday just as you were to look at this? If you think that sending troll emoticons to a commiter of Apache is a good idea just before the release when you know how stressful people in general are I think you pretty much missed the memo about what we are doing here.

Reply:
@driftx are you ok with me doing renaming of other startup checks in this ticket or we should have a dedicated one just for that? I think this might go just as part of this ticket as this whole check stuff we did was never released yet. I do not want to generate more ticket-work.

Reply:
It's not a large change, we can do it here.

Reply:
> If your interest was genuine, you would not sabotage this ticket by your rather remarkably strong preference to include some prefix to a configuration key, even making fun of that. What is funny to you

I’m not sabotaging anything. I’m just wary about introducing cryptic configuration names and trying to be nice about it - since you seem to be inflexible about a trivial review feedback. That’s why I added a smiley face. 😊

Reply:
> That we are not more than two weeks from the freeze and you are sending me trolling faces in the private communication telling me your notebook has broken yesterday just as you were to look at this?

I don’t understand what this is supposed to mean. My computer is indeed broken and I’m writing this from my mobile phone since I feel this seemingly inoffensive feedback could improve the quality and usability of the feature being added.

Reply:
>  If you think that sending troll emoticons to a commiter of Apache is a good idea just before the release when you know how stressful people in general are I think you pretty much missed the memo about what we are doing here.

Are you suggesting me to not use the troll emoji? This is one of my favorite emojis but I will happily no longer use it with you in private communications if you find it offensive.

I am sorry about your stress - I suggest taking some time off to rest and chill since you seem to be taking a candid review feedback personally.

Reply:
No @pauloricardomg , do not make it about myself. I am doing perfectly fine. It is primarily about your toxicity you are spreading around, from my point of view. I think you should not give people false promises for months to take a look at something and then, yes I say it, purposefully delaying it by trolling. And no, it is not about "dude but I am not obliged by anything this is open source I do it in my free time". No. There are millions ways to prevent this but to my very bitter surprise you are fine doing it intentionally.

Reply:
_My computer is indeed broken and I’m writing this from my mobile phone_

@pauloricardomg  Do you believe your own lies? The probability that you 1) have a broken notebook just minutes before reviewing the patch you were promising to review i-do-not-know-how-many-times 2) just when Brandon takes a look you immediately start to respond very elaboratively and promptly 3) sending me trolling messages about that.

Do you think this is a plausible case to support your claim that your intentions are genuine? Could we stop this discussion already to not embarrass you publicly even more you already are?

Reply:
> It is primarily about your toxicity you are spreading around, from my point of view.

I don’t consider making a review commentary “spreading toxicity around”. I believe you are being toxic by publicizing a personal matter instead of trying to settle it privately.

>  I think you should not give people false promises for months to take a look at something and then, yes I say it, purposefully delaying it by trolling.

I’m not purposefully delaying anything nor giving anyone false promises. I volunteered to review your patch, but unfortunately due to unexpected circumstances I was not able to review it as soon as I hoped. I will be more careful next time when providing estimates for review since this is not my primary activity.

> Do you believe your own lies?

I think you start to become disrespectful by calling me a liar - and this breaks Apache’s [code of conduct](https://www.apache.org/foundation/policies/conduct.html):

> Be careful in the words that we choose. Whether we are participating as professionals or volunteers, we value professionalism in all interactions, and take responsibility for our own speech. Be kind to others. Do not insult or put down other participants. Harassment and other exclusionary behaviour are not acceptable.

I have a broken machine and I was not able to complete the review of your patch in a timely manner. I commented on Brandon’s review comment because I had previously reviewed that snippet and I agreed with his comments. I did not send you any trolling message. I was hoping to review your patch by Monday but my computer broke - I put the trolling emoji because I was already embarrassed since the review was already delayed - even though I still planned to review it as I’m doing it now (from mobile).

> Could we stop this discussion already to not embarrass you publicly even more you already are?

I think you are the one that should be embarrassed here by:
1. Taking personal matters publicly
2. Being impolite and disrespectful
3. Breaking apache code of conduct by calling me a liar for making a simple review request that you do not agree with.

Reply:
_Be careful in the words that we choose. Whether we are participating as professionals or volunteers, we value professionalism in all interactions, and take responsibility for our own speech. Be kind to others. Do not insult or put down other participants. Harassment and other exclusionary behaviour are not acceptable._

That is what you are doing by sending me trolling messages and emojis. Nobody asked you to do that. You are the one who started this by not respecting my time and interests and exploting it by trolling and making fun of it. That is the definition of abuse.

Please, do not do it again. This is two way street, Paulo. You will be called on your actions (and inactions) every time.

Reply:
I did not troll or make fun of you. I use emojis to de-escalate tense situations and make communication more human. It seems like you understood this intent incorrectly. I am sorry if this hurt you and I will be more careful when using emojis in textual communication, since these can have different meanings in different cultures.

This does not give you the right to insult me by calling me a liar. You need to act more professionally and be more careful in your choice of words.

Reply:
I decided to have all our conversation from this point onwards public either in jira or on the mailing list. I do not have any interest to talk to you in private until our communication improves.

I will try to wrap this ticket tomorrow.